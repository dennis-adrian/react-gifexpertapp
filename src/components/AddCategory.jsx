import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value);

    }

    const handleSubmit = (e) => {
        //prevenir el comportamiento por defecto del form
        //es decir, prevenir que se recarge
        e.preventDefault();


        if (inputValue.trim().length > 2) {

            //categs representa el valor inicial
            //el valor del arreglo categories
            //que se encuentra en el componoente GifExpertApp
            //y a ese valor le agregamos el input value
            setCategories(categs => [inputValue, ...categs]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>

            <h1>{inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
export default AddCategory;