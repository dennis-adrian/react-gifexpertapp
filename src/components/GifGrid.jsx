import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {/* esto es como un operador ternario
                la ventaja es que solo valida por true
                no es necesario un false */}
            {loading && <p>Loading...</p>}

            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
