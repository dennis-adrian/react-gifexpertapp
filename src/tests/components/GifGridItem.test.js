import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing on <GifGridItem />', () => {

    const title = 'Un t√≠tulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should have a paragrpah with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('should have the image equal to the url and alt of the props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('should have class animate__fadeIn in the div element', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    
})
