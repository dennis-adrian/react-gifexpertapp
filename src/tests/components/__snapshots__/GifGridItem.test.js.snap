// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <GifGridItem /> should the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
